### PR TITLE
[v1.13.x] prov/efa: backport local read and shm fix

### PR DESCRIPTION
**Pull request by wzamazon**

### CODE REVIEW

**Comment by wzamazon** [comment]
bot:aws:retest

**Comment by wzamazon** [comment]
bot:aws:retest

File: prov/efa/src/rxr/rxr_init.c [hunk 1]
```diff
@@ -159,34 +160,44 @@ static void rxr_init_env(void)
...
+	raw_addr = (struct efa_ep_addr *)ptr;
+	if (!inet_ntop(AF_INET6, raw_addr->raw, gidstr, INET6_ADDRSTRLEN)) {
+		FI_WARN(&rxr_prov, FI_LOG_CQ, "Failed to convert GID to string errno: %d\n", errno);
+		return -errno;
 	}
-	qpn = ((struct efa_ep_addr *)addr)->qpn;
-
-	ret = snprintf(smr_name, NAME_MAX, "%ld_%s_%d", (size_t) getuid(), gid, qpn);
 
+	ret = snprintf(smr_name, SMR_NAME_MAX, "%s_%04x_%08x_%04x",
```
Comment (rwespetal) [inline prov/efa/src/rxr/rxr_init.c:199]: `ret` should be checked to make sure it is smaller than `SMR_NAME_MAX`

**Review by rwespetal — APPROVED** [review]

**Comment by wzamazon** [comment]
bot:aws:retest

**Pull request merged**